src/include/cursor_inline.h

```diff
@@ -262,7 +262,7 @@ __cursor_reset(WT_CURSOR_BTREE *cbt)
      * there's a debug mode where an application can force the eviction in order to test or stress
      * the system. Clear the reference so we never try the release twice.
      */
-    if (F_ISSET(cursor, WT_CURSTD_DEBUG_RESET_EVICT))
+    if (F_ISSET(cursor, WT_CURSTD_DEBUG_RESET_EVICT) || evict_page)
```

Comment:
Can we use a flag on cbt so we don't need to pass this bool

Reply:
Updated in [29a2a3a](https://github.com/wiredtiger/wiredtiger/pull/7687/commits/29a2a3a1fd418db9f68692dc60add9c232a54e52)